src/Controller/RentalController.php

```diff
@@ -198,12 +198,15 @@ public function book(Request $request, $rentalId): Response
         $startDate = $startDateInput ? new DateTime($startDateInput) : null;
         $endDate = $endDateInput ? new DateTime($endDateInput) : null;
 
-
-
-
+        if ($startDate > $endDate) {
+            return $this->render('rental/error.html.twig', [
+                'error' => 'Start date cannot be later than end date.'
+            ]);
+        }
         // Calculate total price based on night cost and number of days
         $interval   = $startDate->diff($endDate);
         $days       = $interval->days;
+        $days = $days ?: 1;  
```

Comment:
Suggestion to combine these:

```suggestion
        $days       = $interval->days ?: 1;
```